doc/ins_sign_tx.md

```diff
@@ -71,44 +74,89 @@ Initializes signing request.
 |  P2 | unused |
 | data | see below |
 
-**Data**
+*Data*
 
 |Field| Length | Comments|
 |-----|--------|--------|
 |tx id (hash) | 32 | |
 |output index |  4 | Big endian |
 
 
-### 3 - Set outputs & amounts
+### 3 - Set outputs
 
-**Command**
+For each output, at least two messages are required: the first one with top-level data and the last one for confirmation. The messages in between them describe multiasset tokens if such are included in the output (one message for each asset group, followed by messages for tokens included in the group).
+
+**Command (top-level output data)**
 
 |Field|Value|
 |-----|-----|
 |  P1 | `0x03` |
-|  P2 | unused |
-| data | Tx output, data depending on type |
+|  P2 | `0x30` |
+| data | depends on output type |
 
-**Data for SIGN_TX_OUTPUT_TYPE_ADDRESS**
+*Data for SIGN_TX_OUTPUT_TYPE_ADDRESS_BYTES*
 
 This output type is used for regular destination addresses.
 
 |Field| Length | Comments|
 |-----|--------|--------|
-|Amount| 8| Big endian. Amount in Lovelace|
-|Output type| 1 | `SIGN_TX_OUTPUT_TYPE_ADDRESS=0x01`|
+|Output type| 1 | `SIGN_TX_OUTPUT_TYPE_ADDRESS_BYTES=0x01`|
+|Address size| 4 | Big endian|
 |Address| variable | raw address (before bech32/base58-encoding)|
+|Amount| 8| Big endian. Amount in Lovelace|
+|Number of asset groups| 4 | Big endian|
 
-**Data for SIGN_TX_OUTPUT_TYPE_ADDRESS_PARAMS**
+*Data for SIGN_TX_OUTPUT_TYPE_ADDRESS_PARAMS*
 
 This output type is used for change addresses. Depending (mostly) on staking info, these might or might not be shown to the user. 
 (See [src/securityPolicy.c](../src/securityPolicy.c) for details.)
 
 |Field| Length | Comments|
 |-----|--------|--------|
-|Amount| 8| Big endian. Amount in Lovelace|
 |Output type| 1 | `SIGN_TX_OUTPUT_TYPE_ADDRESS_PARAMS=0x02`|
-|Address params | variable | see `parseAddressParams` in [src/addressUtilsShelley.c](../src/addressUtilsShelley.c)|
+|Address params | variable | see `view_parseAddressParams` in [src/addressUtilsShelley.c](../src/addressUtilsShelley.c)|
+|Amount| 8| Big endian. Amount in Lovelace|
+|Number of asset groups| 4 | Big endian|
+
+**Command (asset group)**
```

Comment:
it's probably worth noting that the asset groups/asset names get serialized into the transaction CBOR in the same order the respective APDUs are passed, so it's up to the integrating party to order them correctly